decisions/2022-12-12_patient_and_program_transfer.md

```diff
@@ -0,0 +1,128 @@
+- How to manage programs at multiple stores and transfer/sync programs between stores?
+- How to manage program permission for different stores?
+
+**Summary of existing mSupply patient editing**:
+
+- Patients are only editable on the home store
+- Only patients from the `name` table can be edited (other names can't be edited). Note, patient push needs to be implemented in omSupply.
+- The patient's home store is identified in the `name` table by the `supplying_store_id` column
+- Its currently not directly supported to change a patient's home store
+
+**Summary of existing mSupply patient search and patient visibility**:
+
+- The endpoint `/patient-search` looks up all patients on central (no access control). The endpoint returns all patient information for a found patient.
+- The client can choose a patient and call `/nsj` to request a new `name_store_join` in order to make the patient visible in a store.
+- The client doesn't wait for sync but directly stores the fetched `name` and `name_store_join` records (client might receive entries again through sync)
+- patient is now visible on the store and syncs to the store (because of the `name_store_join`)
+
+**Transfer Programs Requirements**
+In general, a patient can have a list of programs and each program can be treated at different stores/clinic.
+A patient may change the clinic where the program is treated.
+Thus, we need to be able to record the program store affiliation.
+Potentially a program can be treated at multiple stores.
+It's probably also useful to have a clinic transfer history.
+
+There are three types of transfers:
+
+- External in transfers: program transfer from an external clinic to mSupply
+- External out transfers: program is not manage by mSupply any more
+- Internal transfers, a program is transferred to a different store/clinic within mSupply
+
+**Patient visit**
+A patient might have a program encounter at another store.
+
+Patient Visit Permissions:
+
+- Required patient access during a visit should be minimal.
+- Practitioner needs access to base patient information
+- A visit should show up at the patient's clinic (and resets the "lost to follow up" status)
+
+## Options
+
+### Option 1
+
+**Patient editing:**
+
+- When changing a patient document the patient `name` is updated and pushed to central
+- For this to work we need to allow client to edit patients at stores that are not the patient's home store.
+  (Because there is no fixed store affiliation for programs)
+- When opening a patient document, merge back potential changes from the `name` table to the patient document.
+  This makes it possible to integrate edits of the patient's name table without help from the central server.
+
+**Patient lookup/visibility/sync:**
+
+- Mainly as done in current mSupply
+- Patient documents are synced through the `name_store_join`, i.e. a patient's documents are synced to stores associated with the patient. (in omSupply just trigger sync to fetch patient document data?)
```

Comment:
> in omSupply just trigger sync to fetch patient document data?

Do you mean if patient was just added through lookup ? I guess for now trigger sync should work, as long as mSup adds all patient related data to outgoing sync queue when `/njs` is called

Reply:
yes thats how Chris explained it (if I got it right) otherwise we have to do it the same way we do it in mSupply

Reply:
Cool, yeah I think this would work for now, ideally (maybe for option 2 or option 1 version 2) we would have an endpoint to add `njs` and get all patient data (in the same form as sync) and integrate it